bluesky/plans.py

```diff
@@ -697,8 +697,8 @@ def scan_nd(detectors, cycler, *, per_step=None, md=None):
            }
     _md.update(md or {})
     try:
-        dimensions = [(motor.hints['fields'], 'primary')
-                      for motor in cycler.keys]
+        dimensions = [([motor.hints['fields'] for motor in cycler.keys], 'primary')
```

Comment:
I don't think this is right as it says for a general scan_nd there is only one dimension and it is all of the keys.

In general this should be n-dimensions (so `len(dimensions) > 1`) each of which has 1 fiield.  It should be the job of `inner_product_scan` which knows that there is really only 1 dimension to replace the default / general case with the specialized case.

Reply:
yes this is erroneous and should not have been touched thanks! removed